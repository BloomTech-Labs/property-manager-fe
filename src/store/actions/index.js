// IMPORTS/INITIALIZATION =========================|
// ================================================|
import axios from 'axios';
import axiosAuth from '../../helpers/axiosAuth';
// ------------------------------------------------|
// AUTH ACTIONS ===================================|
// ================================================|
export const AUTH_REQUEST_START = 'AUTH_REQUEST_START';
export const AUTH_REQUEST_SUCCESS = 'AUTH_REQUEST_SUCCESS';
export const AUTH_REQUEST_FAIL = 'AUTH_REQUEST_FAIL';
// ------------------------------------------------|
// LOGIN / SIGNUP ---------------------------------|
export const auth = url => (email, password) => async dispatch => {
  dispatch({ type: AUTH_REQUEST_START });

  try {
    const res = await axios.post(url, {
      email,
      password
    });

    localStorage.setItem('token', res.data.token);

    dispatch({ type: AUTH_REQUEST_SUCCESS, payload: { token: res } });
  } catch (err) {
    console.error(err);
    dispatch({ type: AUTH_REQUEST_FAIL, payload: { errorMessage: err } });
  }
};
// ------------------------------------------------|
// PROPERTY ACTIONS ===============================|
// ================================================|
export const ADD_PROPERTY_START = 'ADD_PROPERTY_START';
export const ADD_PROPERTY_SUCCESS = 'ADD_PROPERTY_SUCCESS';
export const ADD_PROPERTY_FAIL = 'ADD_PROPERTY_FAIL';
export const GET_PROPERTY_START = 'GET_PROPERTY_START';
export const GET_PROPERTY_SUCCESS = 'GET_PROPERTY_SUCCESS';
export const GET_PROPERTY_FAIL = 'GET_PROPERTY_FAIL';
// ------------------------------------------------|
// CREATE A PROPERTY ------------------------------|
export const createProperty = url => property => async dispatch => {
  dispatch({ type: ADD_PROPERTY_START });
  console.log(property);
  try {
    const res = await axios.post(url, { ...property });

    dispatch({
      type: ADD_PROPERTY_SUCCESS,
      payload: {
        created: res
      }
    });
  } catch (err) {
    console.error(err);

    dispatch({ type: ADD_PROPERTY_FAIL, payload: { errorMessage: err } });
  }
};
// ------------------------------------------------|
// GET PROPERTIES ---------------------------------|
export const getProperties = url => async dispatch => {
  dispatch({ type: GET_PROPERTY_START });

  try {
    const res = await axiosAuth().get(url);

    dispatch({
      type: GET_PROPERTY_SUCCESS,
      payload: {
        properties: res.data
      }
    });
  } catch (err) {
    console.error(err);

    dispatch({ type: GET_PROPERTY_FAIL, payload: { errMsg: err } });
  }
};
// ------------------------------------------------|
// USER ACTIONS ===================================|
// ================================================|

// define your user actions here

// ------------------------------------------------|
