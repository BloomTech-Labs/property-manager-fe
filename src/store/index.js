import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { getFirebase } from 'react-redux-firebase';
import reducer from './reducers';

const middleware =
  process.ENV === 'production'
    ? [thunk.withExtraArgument(getFirebase)]
    : [thunk.withExtraArgument(getFirebase), logger];

export const makeStore = state =>
  createStore(
    reducer,
    state,
    compose(
      applyMiddleware(...middleware),

      /* eslint-disable */
      window.__REDUX_DEVTOOLS_EXTENSION__
        ? window.__REDUX_DEVTOOLS_EXTENSION__()
        : f => f
    )
  );

export default makeStore();
