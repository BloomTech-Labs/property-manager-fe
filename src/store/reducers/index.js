import { combineReducers } from 'redux';
import authReducer from './authReducer';
import propReducer from './propReducer';
import getUserReducer from './getUserReducer';
import workOrderReducer from './workOrderReducer';

export default combineReducers({
  authReducer,
  propReducer,
  getUserReducer,
  workOrderReducer
});
