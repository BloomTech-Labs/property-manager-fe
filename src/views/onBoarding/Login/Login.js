import React from 'react';
import LogInForm from '../../../components/Auth/LoginForm/LoginForm';
import '../../../scss/components/_onboardingForms.scss';

const Login = () => {
  return (
    <div className="main-form-page-wrapper" data-testid="LoginForm">
      <LogInForm />
    </div>
  );
};

export default Login;
