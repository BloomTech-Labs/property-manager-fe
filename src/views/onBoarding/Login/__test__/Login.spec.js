import React from 'react';
import * as rtl from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../store';
import Login from '../Login';

afterEach(rtl.cleanup);

describe('SignUp Form tests...', () => {
  it('renders the form...', () => {
    const { getByTestId } = rtl.render(
      <Provider store={store}>
        <Login />
      </Provider>
    );

    const Form = getByTestId('LoginForm');

    expect(Form).toBeTruthy();
  });

  it('snapshot', () => {
    const { container } = rtl.render(
      <Provider store={store}>
        <Login />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('login a user with creds', async () => {
    const handleSubmit = jest.fn();

    const fakeUser = {
      email: 'test@test.com',
      password: 'password1'
    };

    const { getByPlaceholderText, getByTestId } = await rtl.render(
      <Provider store={store}>
        <Login onSubmit={handleSubmit} />
      </Provider>
    );

    const email = getByPlaceholderText('Enter your email address');
    const password = getByPlaceholderText('Type your password');
    const submitBtn = getByTestId('submit-btn');

    email.value = fakeUser.email;
    password.value = fakeUser.password;
    submitBtn.click();

    expect(handleSubmit).toHaveBeenCalledTimes(0);
  });
});
