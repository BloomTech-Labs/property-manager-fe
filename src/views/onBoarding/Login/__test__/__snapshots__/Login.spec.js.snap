// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Form tests... snapshot 1`] = `
<div
  class="main-form-page-wrapper"
  data-testid="LoginForm"
>
  <div
    class="LoginForm"
  >
    <div
      class="form-wrapper"
    >
      <h2>
        Login
      </h2>
      <form
        action="#"
        class="form-element"
        data-testid="form-element"
      >
        <div
          class="input-wrapper"
        >
          <label
            for="email"
          >
            <span
              class="css-1bcyk0w"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </span>
             
            Email
          </label>
          <input
            name="email"
            placeholder="Enter your email address"
            type="email"
            value=""
          />
        </div>
        <div
          class="input-wrapper"
        >
          <label
            for="Password"
          >
            <span
              class="css-1bcyk0w"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </span>
             
            Password
          </label>
          <input
            name="password"
            placeholder="Type your password"
            type="password"
            value=""
          />
        </div>
        <div
          class="submit-btn-wrapper"
        >
          <button
            class="btn"
            data-testid="submit-btn"
            type="submit"
          >
            Submit
          </button>
          <a
            href="/iamareallyforgetfulperson"
          >
            Forgot your password?
          </a>
        </div>
        <div
          class="css-1glmex"
        >
          <p
            class="css-1tqdaz3"
          >
            Don't have an account?
          </p>
          <button
            class="flip"
            type="button"
          >
            Sign Up
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
